### PR TITLE
explain Available condition behavior in deployment concepts

### PR DESCRIPTION
**Pull request by atiratree**
We need to come to a consensus how to define an Available condition. This has been already discussed in these places

- https://github.com/kubernetes/kubernetes/pull/113228
- https://github.com/kubernetes/enhancements/pull/3524
-  SIG Apps 20221017 https://www.youtube.com/watch?v=Td0tHnIsNpo

cc @smarterclayton @soltysh  @janetkuo @alculquicondor @ravisantoshgudimetla

### CODE REVIEW

File: content/en/docs/concepts/workloads/controllers/deployment.md [hunk 1]
```diff
@@ -1027,8 +1027,23 @@ Conditions:
   Progressing   True    NewReplicaSetAvailable
 ```
 
-`type: Available` with `status: "True"` means that your Deployment has minimum availability. Minimum availability is dictated
-by the parameters specified in the deployment strategy. `type: Progressing` with `status: "True"` means that your Deployment
+`type: Available` with `status: "True"` means that your Deployment is healthy and has a minimum availability (`.status.availableReplicas` is at least equal to `.spec.replicas  - maxUnavailable`).
+`maxUnavailable` is computed according to the deployment strategy:
+- For the `RollingUpdate` strategy, it is a resolved value of `.spec.strategy.rollingUpdate.maxUnavailable`.
+- For the `Recreate` strategy, it is always equal to 100% of the replicas that we allow to be down, so that new rollouts can proceed.
```
Comment (atiratree) [inline content/en/docs/concepts/workloads/controllers/deployment.md:1033]: This counts with a change proposed in https://github.com/kubernetes/kubernetes/pull/113228. Currently it is 0.

File: content/en/docs/concepts/workloads/controllers/deployment.md [hunk 2]
```diff
@@ -1027,8 +1027,23 @@ Conditions:
...
+
+This condition should always have `status: "True"` in day-to-day operations; during and also after a rollout is completed.
+The condition will become `status: "False"` if there is an abnormal amount of disrupted pods (higher than a resolved value of `maxUnavailable`).
+
+{{< note >}}
+The `Available` condition is designed to support rollouts first. If `maxUnavailable` is equal to `.spec.replicas`, 
+then this condition will never become `status: "False"`, even if no pods are running. 
+In this case you should check `.status.availableReplicas` to be sure that at least 1 pod is present. 
+It is recommended to have the number of `.spec.replicas` higher than `maxUnavailable`, to have always some pods available during a rollout.
```
Comment (smarterclayton) [inline content/en/docs/concepts/workloads/controllers/deployment.md:1042]: Good description!

**Comment by dims** [comment]
- looks like https://github.com/kubernetes/website/pull/39892 is waiting for this to merge first
- also this one is waiting on https://github.com/kubernetes/kubernetes/pull/113228

**Comment by smarterclayton** [comment]
I would prefer to get the update to the strategy - I think we made the case in 113228 that we should be consistent and the current behavior provides no value for end users on recreate (they can only be wrong).

**Comment by kbhawkey** [comment]
👋 @atiratree .
If you are waiting on code changes, should the PR be held until the changes are completed?

**Comment by kbhawkey** [comment]
Hello @atiratree .
What's happening with these changes?

**Comment by atiratree** [comment]
@smarterclayton would you have time to revisit https://github.com/kubernetes/kubernetes/pull/113228? Or should I overtake it, or try to wrap this one first?

**Comment by divya-mohan0209** [comment]
Hello @atiratree , please could you advise where we are with these changes?

**Comment by divya-mohan0209** [comment]
Hello @atiratree , please could you advise where we are with these changes?

**Comment by sftim** [comment]
There hasn't been an update in a long while.

I will close this PR.
@atiratree - your branch is still there, and you are welcome to make a new PR based on the work you have already done.

/close

**Pull request closed without merging**